### PR TITLE
RavenDB-16367 implemented LockMode in DatabaseRecord

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 1]
```diff
@@ -717,48 +721,61 @@ public async Task Delete()
...
+                            if (rawRecord == null)
+                                continue;
+
+                            switch (rawRecord.LockMode)
                             {
-                                if (rawRecord == null)
+                                case DatabaseLockMode.Unlock:
+                                    databasesToDelete.Add(databaseName);
+                                    break;
+                                case DatabaseLockMode.PreventDeletesIgnore:
```
Comment (ayende) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs:744]: Add a note to the log? Maybe an alert?
For that matter, if this is set, we probably want to record who is trying the delete (ip, cert, etc, no)?

Comment (ppekrol) [reply src/Raven.Server/Web/System/AdminDatabasesHandler.cs:744]: We are recording that beforehand (auditlog). Maybe we actually should record that after (or both?) ?

Comment (ayende) [reply src/Raven.Server/Web/System/AdminDatabasesHandler.cs:744]: Audit log may not be configured.

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 2]
```diff
@@ -1393,6 +1419,51 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
...
+                        databasesToUpdate.Add(databaseName);
+                    }
+                }
+
+                if (databasesToUpdate.Count > 0)
+                {
+                    long index = 0;
+                    foreach (var databaseName in databasesToUpdate)
+                    {
+                        //var result = await ServerStore.WriteDatabaseRecordAsync(kvp.DatabaseRecord.DatabaseName, kvp.DatabaseRecord, kvp.Index, raftRequestId);
```
Comment (ml054) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: remove?

**Comment by ml054** [comment]
@ppekrol I'm waiting for icons - once I get them will apply them and it will ready for merge.

**Comment by ml054** [comment]
@ppekrol icons are in place so everything is ready on my side.

**Pull request merged**